### PR TITLE
Update Apache Maven to 3.8.8 and Maven Wrapper to 3.2.0

### PR DESCRIPTION
**Pull request by jorsol**

### CODE REVIEW

**Review by aloubyansky — APPROVED** [review]
Thanks @jorsol 
@ia3andy do you mind giving it a look too? Thanks!

File: extensions/resteasy-classic/rest-client/runtime/src/test/java/io/quarkus/restclient/runtime/RestClientBaseTest.java [hunk 1]
```diff
@@ -39,7 +40,7 @@ public class RestClientBaseTest {
     private static final String TRUSTSTORE_PASSWORD = "truststorePassword";
     private static final String KEYSTORE_PASSWORD = "keystorePassword";
 
-    @TempDir
+    @TempDir(cleanup = CleanupMode.NEVER)
```
[the remark this answers is not in the document]
Comment (aloubyansky) [reply extensions/resteasy-classic/rest-client/runtime/src/test/java/io/quarkus/restclient/runtime/RestClientBaseTest.java]: Thanks @jorsol

**Review by gsmet — APPROVED** [review]

**Comment by famod** [comment]
@aloubyansky continuing the discussion from https://github.com/quarkusio/quarkus/pull/31917#issuecomment-1473558774, I suppose that beanbag change (and Maven 3.9.1) will not be backported to 2.16? If so, I was thinking that maybe this PR here (Maven 3.8.8) should be merged first so that it can ve backported to 2.16.
WDYT? Also cc @gsmet

**Comment by aloubyansky** [comment]
I wanted to discuss this with @gsmet i thought we should backport the 3.9 support to 2.16. But we need to see what the actual change will be. I don't mind merging this one first, btw.

**Comment by ia3andy** [comment]
@aloubyansky looks good, sorry for the delay 😅

**Pull request merged**